sysbuild/CMakeLists.txt

```diff
@@ -333,11 +333,20 @@ function(${SYSBUILD_CURRENT_MODULE_NAME}_pre_cmake)
           set_config_bool(mcuboot CONFIG_NORDIC_QSPI_NOR_XIP y)
         endif()
       else()
-        set(${DEFAULT_IMAGE}_SIGNING_SCRIPT "${ZEPHYR_NRF_MODULE_DIR}/cmake/sysbuild/image_signing.cmake" CACHE INTERNAL "MCUboot signing script" FORCE)
-
-        if(SB_CONFIG_MCUBOOT_BUILD_DIRECT_XIP_VARIANT)
-          set(mcuboot_secondary_app_SIGNING_SCRIPT "${ZEPHYR_NRF_MODULE_DIR}/cmake/sysbuild/image_signing.cmake" CACHE INTERNAL "MCUboot signing script" FORCE)
-        elseif(SB_CONFIG_MCUBOOT_MODE_FIRMWARE_UPDATER)
+        get_property(images GLOBAL PROPERTY sysbuild_images)
+        foreach(image ${images})
+          # Do not merge/change signing method for MCUboot image.
+          if("${image}" STREQUAL "mcuboot")
+            continue()
+          endif()
+          # Do not merge/change signing method for MCUboot secondary (variant) image.
+          if("${image}" STREQUAL "s1_image")
+            continue()
+          endif()
+          set(${image}_SIGNING_SCRIPT "${ZEPHYR_NRF_MODULE_DIR}/cmake/sysbuild/image_signing.cmake" CACHE INTERNAL "MCUboot signing script" FORCE)
```

Comment:
this should only apply to the variant images, this blindly applies it to everything, fix (use a list or property or similar to know what images to apply to)

Reply:
Looking at the existing code: this should apply to both main and variant images, not only to the variants.
Reworked to use lists stored as global properties.